test/unit_tests/apps/owning-a-home/js/explore-rates/RateCheckerChartMenu-spec.js

```diff
@@ -0,0 +1,155 @@
+);
+
+const simulateEvent =
+  require( '../../../../../util/simulate-event' ).simulateEvent;
+
+const STATES = RateCheckerChartMenu.STATES;
+
+let highCharts;
+let chartMenu;
+let chartMenuDOM;
+let chartMenuBtnDOM;
+
+const HTML_SNIPPET = `
+  <section id="chart-section" class="chart">
+  </section>
+    <div class="chart-menu">
+      <button class="chart-menu_btn
+                     cf-icon
+                     cf-icon__after
+                     cf-icon-down">
+        Download chart
+      </button>
+      <ul class="chart-menu_options">
+        <li data-export-type="0">PNG</li>
+        <li data-export-type="1">SVG</li>
+        <li data-export-type="2">JPEG</li>
+        <li data-export-type="3">Print chart</li>
+      </ul>
+    </div>
+`;
+
+describe( 'explore-rates/RateCheckerChartMenu', () => {
+  beforeEach( () => {
+    document.body.innerHTML = HTML_SNIPPET;
+
+    highCharts = new Highcharts.Chart( 'chart-section', {
+      xAxis: {
+        categories: [ 'Jan', 'Feb' ]
+      },
+      series: [ { data: [ 29.9, 71.5 ] } ]
+    } );
+
+    highCharts.exportChart = jest.fn();
+    highCharts.print = jest.fn();
+
+    chartMenu = new RateCheckerChartMenu( highCharts );
+    chartMenuDOM = document.querySelector( '.chart-menu' );
+    chartMenuBtnDOM = document.querySelector( '.chart-menu_btn' );
+  } );
+
+  describe( 'new RateCheckerChartMenu()', () => {
+    it( 'should set the state to closed', () => {
+      expect( chartMenu.state ).toEqual( { position: STATES.CLOSED } );
+    } );
+
+    it( 'should set a reference to the highcharts instance', () => {
+      expect( highCharts ).toEqual( chartMenu.highCharts );
+    } );
+
+    it( 'should set a reference to the base element', () => {
+      expect( chartMenuDOM.isEqualNode( chartMenu.element ) ).toBe( true );
+    } );
+
+    it( 'should initialize the click event on the menu element', () => {
+      const _initEvents = RateCheckerChartMenu.prototype._initEvents;
+      const mockInitEvents = jest.fn();
+      RateCheckerChartMenu.prototype._initEvents = mockInitEvents;
+      const chartMenu = new RateCheckerChartMenu( highCharts );
+
+      expect( mockInitEvents.mock.calls.length ).toBe( 1 );
+      RateCheckerChartMenu.prototype._initEvents = _initEvents;
+    } );
+  } );
+
+  describe( 'open()', () => {
+    it( 'should set the open state', () => {
+      chartMenu.open();
+      expect( chartMenu.state ).toEqual( { position: STATES.OPEN } );
+    } );
+  } );
+
+  describe( 'close()', () => {
+    it( 'should set the closed state', () => {
+      chartMenu.close();
+      expect( chartMenu.state ).toEqual( { position: STATES.CLOSED } );
+    } );
+  } );
+
+  describe( 'render()', () => {
+    it( 'should set the proper classes on the menu DOM', () => {
+      chartMenu.open();
+      expect( chartMenuDOM.classList.contains( 'chart-menu__open' ) )
+        .toEqual( true );
+    } );
+  } );
+
+  describe( '_setState()', () => {
```

Comment:
Should `setState` be part of the public API or should `_setState` not be inside the instance?

Reply:
So I prepended the underscore to denote the fact that it shouldn't be called unless really necessary.

Reply:
Yeah I get that. I'd probably not test anything but the public API. The private API should be covered through the functionality of the public behavior. If you don't want it accessible at all, it could be moved outside of the class to something like:

```js
_setState( inst, state = {} ) {
    const oldState = assign( {}, inst.state );
    assign( inst.state, state );
    inst.render( oldState, state );
}
```

Reply:
@anselmbradford, I'm borrowing the from the python convention: https://shahriar.svbtle.com/underscores-in-python. Whether or not it makes sense for it to be a private member and on the prototype we should discuss.